rfcs/20230818-Dropout/README.md

```diff
@@ -0,0 +1,79 @@
+# Introducing Dropout primitive attribute
+
+## Introduction
+
+In many DNN and GNN models, [Dropout](https://en.wikipedia.org/wiki/Convolutional_neural_network#Dropout) is used to improve training results. In some cases, this layer can take a significant amount of time. To enhance the performance of training, we want to optimize it and, as a result, fuse it with the previous primitive.
+
+This idea was [proposed](https://github.com/oneapi-src/oneDNN/pull/760) some time ago. As a result of that discussion, a primitive attribute was proposed to resolve the performance gap.
+
+## Proposal
+
+Additional function to set dropout attibute in C API:
+
+```c
+/// Returns the parameters of a drop out attribute.
+///
+/// @param attr Primitive attributes.
+/// @param enable_drop Drop-out enable/disable flag
+/// @param mask_desc Output memory descriptor of a drop out mask.
+/// @returns #dnnl_success on success and a status describing the error
+///     otherwise.
+dnnl_status_t DNNL_API dnnl_primitive_attr_get_dropout(
```

Comment:
Why is Droppout added as an attribute and not a post operation?
For post operations there is more flexibility in case we need to adjust order in which operations are applied after the main operation.

Reply:
The suggestion to use attribute mainly comes from complex primitives, like RNN, where post-op semantics is not well defined, and for which we already had requests in the past for fused dropout.

Reply:
> The suggestion to use attribute mainly comes from complex primitives, like RNN, where post-op semantics is not well defined, and for which we already had requests in the past for fused dropout.

O, RNN... Makes sense. @itaraban Could you please add this information somewhere in the RFC?

Reply:
@mgouicem , thanks!
@igorsafo , I updated README